src/react-integration/hooks/useResourceNew.ts

```diff
@@ -55,18 +42,27 @@ function useManyResources<A extends ResourceArgs<any, any, any>[]>(
       Params extends Readonly<object>,
       Body extends Readonly<object | string> | void,
       S extends Schema
-    >([select, params]: ResourceArgs<S, Params, Body>) =>
+    >([fetchShape, params]: ResourceArgs<S, Params, Body>) =>
       // eslint-disable-next-line react-hooks/rules-of-hooks
-      useCacheNew(select, params),
+      useCacheNew(fetchShape, params),
   );
   const promises = resourceList
-    .map(([select, params]) =>
+    .map(([fetchShape, params]) =>
       // eslint-disable-next-line react-hooks/rules-of-hooks
-      useRetrieve(select, params),
+      useRetrieve(fetchShape, params),
     )
     // only wait on promises without results
     .map((p, i) => !hasUsableData(resources[i], resourceList[i][0]) && p);
 
+  // throw first valid error
+  for (let i = 0; i < resourceList.length; i++) {
+    const [fetchShape, params] = resourceList[i];
+    const resource = resources[i];
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const error = useError(fetchShape, params, resource);
+    if (error && !promises[i]) throw error;
```

Comment:
this logic is now identical to the singular case, since we store the same conditional in this map.